### PR TITLE
remove user chapter, replace with user chapter role

### PR DESCRIPTION
**Pull request by timmyichen**
- [x] I have read [Chapter's contributing guidelines](https://github.com/freeCodeCamp/chapter/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update README.md`).
- [x] My pull request targets the `master` branch of Chapter.

This PR:
* Removes the `UserChapter` model
* Adds factories/seeds/etc for `UserChapterRole` model

Strangely enough, I can't seem to get a migration running for this.  Any ideas?

```
tim@dropten:~/chapter$ npm run typeorm migration:generate -- --name=RemoveChapterUser

> chapter@0.0.1 typeorm /home/tim/chapter
> ts-node -P tsconfig.server.json -r tsconfig-paths/register node_modules/typeorm/cli.js "migration:generate" "--name=RemoveChapterUser"

No changes in database schema were found - cannot generate a migration. To create a new empty migration use "typeorm migration:create" command
```

### CODE REVIEW

**Comment by Zeko369** [comment]
I think I know that the problem is. TypeORM is really finicky when it comes to changes relations. So when you change the migration table it doesn't actually change the migration.

Try creating a new migration for a slightly different column eg `users123`, run the migrations, and then manually change stuff around in the migration file.

**Comment by allella** [comment]
To summarize for myself, and for anybody else watching, it sounds like these changes are furthering the work of #335 and #345 and endeavoring to create a migration to make the schema changes which were previously only part of the models? Is that correct?

Regarding automatically updating the schema image, we have that in progress on #54 and https://github.com/freeCodeCamp/chapter/issues/360#issuecomment-598486978

**Comment by Zeko369** [comment]
Any progress with this @timmyichen?

File: server/models/UserChapterRole.ts [hunk 1]
```diff
@@ -23,15 +24,25 @@ export class UserChapterRole {
   @JoinColumn({ name: 'chapter_id' })
   chapter!: Chapter;
 
+  /*
+    This indicates whether he use wants to receive notifications about this chapter.
```
Comment (Zeko369) [inline server/models/UserChapterRole.ts:28]: Maybe rename this to notifications?
This sounds a lot like someone wanting to join a chapter

Comment (Zeko369) [reply server/models/UserChapterRole.ts:28]: Honestly this is fine as is, we can change it later if others find it confusing

**Pull request merged**